### PR TITLE
[Feature] add blockchain_alias blockchain-spec

### PR DESCRIPTION
**Pull request by boulix3**
Hello this Pull Request adds the "blockchain_alias" option in the blockchainSpecs parameter.

It calls the Admin.AliasChain api when the custom blockchain is ready.

With the timestampvm example, using the timestamp alias, you can use 127.0.0.1:9650/ext/bc/timestamp instead of 127.0.0.1:9650/ext/bc/E8isHenre76NMxbJ3munSQatV8GoQ4XKWQg9vD34xMBqEFJGf.

### CODE REVIEW

**Review by dasconnor — COMMENTED** [review]
I think this looks good but want @felipemadero to take a look

File: local/blockchain.go [hunk 1]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
 	if err := ln.waitForCustomChainsReady(ctx, chainInfos); err != nil {
 		return err
 	}
+
+	ln.RegisterBlockchainAliases(ctx, chainInfos, chainSpecs)
```
Comment (felipemadero) [inline local/blockchain.go:94]: this needs to check for error and return such is that is the case

File: local/blockchain.go [hunk 2]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
+
 	return nil
 }
 
+// if alias is defined in blockchain-specs, registers an alias for the previously created blockchain
+func (ln *localNetwork) RegisterBlockchainAliases(
+	ctx context.Context,
+	chainInfos []blockchainInfo,
+	chainSpecs []network.BlockchainSpec,
+) {
```
Comment (felipemadero) [inline local/blockchain.go:104]: should return error if aliasing operation fails

File: local/blockchain.go [hunk 3]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
+	fmt.Println()
+	ln.log.Info(logging.Blue.Wrap(logging.Bold.Wrap("Registring blockchain aliases...")))
+	for i, chainSpec := range chainSpecs {
+		if chainSpec.BlockchainAlias != nil {
```
Comment (felipemadero) [inline local/blockchain.go:108]: please use this as non optional and compare agains `""`

File: local/blockchain.go [hunk 4]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
 }
 
+// if alias is defined in blockchain-specs, registers an alias for the previously created blockchain
+func (ln *localNetwork) RegisterBlockchainAliases(
+	ctx context.Context,
+	chainInfos []blockchainInfo,
+	chainSpecs []network.BlockchainSpec,
+) {
+	fmt.Println()
+	ln.log.Info(logging.Blue.Wrap(logging.Bold.Wrap("Registring blockchain aliases...")))
```
Comment (felipemadero) [inline local/blockchain.go:106]: Registering

File: local/blockchain.go [hunk 5]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
+	for i, chainSpec := range chainSpecs {
+		if chainSpec.BlockchainAlias != nil {
+			blockchainAlias := *chainSpec.BlockchainAlias
+			chainId := chainInfos[i].blockchainID.String()
+			for nodeName, node := range ln.nodes {
+				ln.log.Info("Registring blockchain alias",
```
Comment (felipemadero) [inline local/blockchain.go:112]: Registering

Comment (felipemadero) [reply local/blockchain.go:112]: would prefer only one log for all 5 nodes, telling the specific alias information to set (that is, a log right before the for)

File: local/blockchain.go [hunk 6]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
+					zap.String("alias", blockchainAlias),
+					zap.String("chain-id", chainId),
+					zap.String("node-name", nodeName))
+				err := node.client.AdminAPI().AliasChain(ctx, chainId, blockchainAlias)
+				if err != nil {
+					ln.log.Error("Error while registring blockchain alias",
```
Comment (felipemadero) [inline local/blockchain.go:118]: this error should be returned and not logged

**Review by felipemadero — COMMENTED** [review]
Left some comments for improvement

File: local/blockchain.go [hunk 7]
```diff
@@ -90,9 +90,42 @@ func (ln *localNetwork) CreateBlockchains(
...
+	for i, chainSpec := range chainSpecs {
+		if chainSpec.BlockchainAlias != nil {
+			blockchainAlias := *chainSpec.BlockchainAlias
+			chainId := chainInfos[i].blockchainID.String()
+			for nodeName, node := range ln.nodes {
+				ln.log.Info("Registring blockchain alias",
+					zap.String("alias", blockchainAlias),
+					zap.String("chain-id", chainId),
+					zap.String("node-name", nodeName))
+				err := node.client.AdminAPI().AliasChain(ctx, chainId, blockchainAlias)
```
Comment (holisticode) [inline local/blockchain.go:116]: Very minor nit: We use the pattern
`if err :=....; err != nil {}` when only an `err` is returned

**Comment by boulix3** [comment]
Awesome, thanks for the feedback.

**Review by felipemadero — COMMENTED** [review]
LGTM. Will approve once @holisticode nit is addressed

**Comment by boulix3** [comment]
> 
Oh right, I didn't see this one. Fixing right now.

**Review by felipemadero — APPROVED** [review]

**Pull request merged**